internal/controller/pipeline_controller.go

```diff
@@ -101,110 +95,84 @@ func (r *PipelineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
-	for _, vector := range vectorInstances {
-		if vector.DeletionTimestamp != nil {
-			continue
-		}
+	eg := errgroup.Group{}
 
-		// Init Controller for Vector Agent
-		vaCtrl := vectoragent.NewController(vector, r.Client, r.Clientset)
-		vaCtrl.SetDefault()
-		// Get Vector Config file
-		byteConfig, err := config.BuildByteConfig(vaCtrl, pipelineCR)
-		if err != nil {
-			if err := pipeline.SetFailedStatus(ctx, r.Client, pipelineCR, err.Error()); err != nil {
-				return ctrl.Result{}, err
+	for _, vector := range vectorAgents {
+		eg.Go(func() error {
+			vaCtrl := vectoragent.NewController(vector, r.Client, r.Clientset)
+			byteConfig, err := config.BuildByteConfig(vaCtrl, pipelineCR)
+			if err != nil {
+				return fmt.Errorf("%w: %w", ErrBuildConfigFailed, err)
 			}
-			if err = pipeline.SetLastAppliedPipelineStatus(ctx, r.Client, pipelineCR); err != nil {
-				return ctrl.Result{}, err
+
+			vaCtrl.Config = byteConfig
+			configCheck := configcheck.New(
+				vaCtrl.Config,
+				vaCtrl.Client,
+				vaCtrl.ClientSet,
+				vaCtrl.Vector,
+				r.ConfigCheckTimeout,
+				configcheck.ConfigCheckInitiatorPipieline,
+			)
+
+			reason, err := configCheck.Run(ctx)
+			if reason != "" {
+				return errors.New(reason)
 			}
+			return err
+		})
+	}
+
+	if err = eg.Wait(); err != nil {
```

Comment:
It is necessary to process a case where the VectorPipeline is considered valid for one vector but not for another. In this situation, we will simply see an error, even though the VectorPipeline may work in some contexts. It might be necessary to change the status structure for the VectorPipeline.

Reply:
I have added information about the agent where the check error occurred in the error reason field. Is this solution satisfactory?